topics/assembly/tutorials/assembly-decontamination/tutorial.md

```diff
@@ -0,0 +1,324 @@
+>
+> 1. Create a new history for this tutorial
+> 2. Import the files from [Zenodo]({{ page.zenodo_link }}):
+>
+>    ```
+>    https://zenodo.org/records/13332983/files/Contaminated_assembly.fasta?download=1
+>    ```
+>    {% snippet faqs/galaxy/datasets_import_via_link.md %}
+>
+> 3. Rename the datasets : `Contaminated Assembly`
+> 4. Check that the datatype is `fasta`
+>
+>    {% snippet faqs/galaxy/datasets_change_datatype.md datatype="fasta" %}
+>
+>
+{: .hands_on}
+
+# Genome Masking
+
+Genome masking is the technique of hiding the region of the genome that cary low complexity, such as repeats. Masking a genome reduces the noise and improves the efficacy of the decontamination by emphasizing the regions that contribute the most to the classification ({% cite saini2016gene %}, {% cite maskedref-website %}).
+The first step of our analysis is therefore to mask our genome assembly. There is two type of masking, depending on the requirements of the tools we are using.
+
+- **Soft Masking** The masked regions are replaced by lower case letters
+- **Hard Masking** The masked regions are replaced by a wildcard letter:  `N`s for nucleotide sequences, `X`s for proteins
+
+
+> <details-title> Genome masking for everyone </details-title>
+>
+> Repeat masking is important for a lot of analyses like sequence alignments, classification, gene annotation. Repetitive and low complexity regions cause problems for search and clustering algorithms based on patterns. For gene annotation, repeats can cause non-specific gene hits.
+>
+{: .details}
+
+In this tutorial we will use Kraken2 to classify our scaffolds, and DustMasker to mask our genome. Dustmasker uses soft masking, while Kraken2 uses Hard masking, so we will do some file manipulations along the way.
+
+The first step is to ensure the sequences in our assembly are in upper cases.
+
+
+> <hands-on-title> Convert our assembly file to upper case </hands-on-title>
+>
+> 1. {% tool [Text transformation](toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sed_tool/1.1.1) %} with the following parameters:
+>    - {% icon param-file %} *"File to process"*: `Contaminated_assembly`
+>    - *"SED Program"*: `s/.*/\U&/` (Convert all letters to uppercase)
+>
+>    > <comment-title> Unsure about your regex? </comment-title>
+>    >
+>    > Ask ChatGPT about it. E.g. [What is the sed command to convert a text to upper case?](https://chatgpt.com/share/de569538-0e4d-41ac-8380-2d52af62b3d7)
+>    {: .comment}
+>
+{: .hands_on}
+
+
+
+> <hands-on-title> Mask the Genome with Dustmasker </hands-on-title>
+>
+> 1. {% tool [NCBI BLAST+ dustmasker](toolshed.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus/ncbi_dustmasker_wrapper/2.14.1+galaxy1) %} with the following parameters:
+>    - *"Subject database/sequences"*: `FASTA file from your history`
+>        - {% icon param-file %} *"Nucleotide FASTA subject file to use instead of a database"*: `output` (output of **Text transformation** {% icon tool %})
```

Comment:
we have run text 'transformation' twice at this point, which should we use? maybe clarify by having users rename the outputs of the tools?